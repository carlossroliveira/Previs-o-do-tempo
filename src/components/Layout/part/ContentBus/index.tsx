// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { Button } from '../../../Button';
import { Input } from '../../../Input';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './contentBusStyles';

export const ContentBus = (): JSX.Element => {
  return (
    <ContainerSC>
      <Input />
      <Button />
    </ContainerSC>
  );
};
