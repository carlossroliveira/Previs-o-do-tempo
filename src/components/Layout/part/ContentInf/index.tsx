// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { SwitchComponent } from '../../../SwitchComponent';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, TitleSC } from './contentInfStyles';

export const ContentInf = (): JSX.Element => {
  return (
    <ContainerSC>
      <TitleSC>Tema</TitleSC>
      <SwitchComponent />
    </ContainerSC>
  );
};
