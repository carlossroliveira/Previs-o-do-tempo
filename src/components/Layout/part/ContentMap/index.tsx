// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------

// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './ContentMapStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const ContentMap = (): JSX.Element => {
  return <ContainerSC>MAPA</ContainerSC>;
};
