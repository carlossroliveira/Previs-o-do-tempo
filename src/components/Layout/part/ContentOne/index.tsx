// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------

// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './contentOneStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const ContentOne = (): JSX.Element => {
  return <ContainerSC>CONTENTONE</ContainerSC>;
};
