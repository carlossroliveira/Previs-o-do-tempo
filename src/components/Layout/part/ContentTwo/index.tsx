// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------

// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './contentTwoStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const ContentTwo = (): JSX.Element => {
  return <ContainerSC>CONTENTTWO</ContainerSC>;
};
