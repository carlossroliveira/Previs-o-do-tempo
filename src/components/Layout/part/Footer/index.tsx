// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------

// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './footerStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const Footer = (): JSX.Element => {
  return (
    <ContainerSC>
      Animações Animações Animações Animações Animações Animações Animações
      Animações Animações Animações Animações Animações Animações Animações
      Animações Animações Animações Animações Animações Animações Animações
      Animações Animações Animações Animações Animações Animações Animações
      Animações Animações Animações Animações Animações Animações Animações
      Animações Animações Animações Animações Animações Animações{' '}
    </ContainerSC>
  );
};
