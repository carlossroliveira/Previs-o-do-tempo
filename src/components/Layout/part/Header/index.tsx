// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------

// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './headerStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const Header = (): JSX.Element => {
  return <ContainerSC>HEADER</ContainerSC>;
};
