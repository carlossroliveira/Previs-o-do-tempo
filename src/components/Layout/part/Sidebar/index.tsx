// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------

// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './sidebarStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const Sidebar = (): JSX.Element => {
  return (
    <ContainerSC>
      <ul>
        <li>LISTA DAS CAPITAIS</li>
        <li>LISTA DAS CAPITAIS</li>
        <li>LISTA DAS CAPITAIS</li>
        <li>LISTA DAS CAPITAIS</li>
        <li>LISTA DAS CAPITAIS</li>
        <li>LISTA DAS CAPITAIS</li>
        <li>LISTA DAS CAPITAIS</li>
        <li>LISTA DAS CAPITAIS</li>
        <li>LISTA DAS CAPITAIS</li>
        <li>LISTA DAS CAPITAIS</li>
        <li>LISTA DAS CAPITAIS</li>
        <li>LISTA DAS CAPITAIS</li>
        <li>LISTA DAS CAPITAIS</li>
        <li>LISTA DAS CAPITAIS</li>
        <li>LISTA DAS CAPITAIS</li>
        <li>LISTA DAS CAPITAIS</li>
        <li>LISTA DAS CAPITAIS</li>
        <li>LISTA DAS CAPITAIS</li>
      </ul>
    </ContainerSC>
  );
};
