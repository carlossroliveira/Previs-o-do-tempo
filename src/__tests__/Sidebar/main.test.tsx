// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { Sidebar } from '../../components/Layout/part/Sidebar';
// -------------------------------------------------
// Utils
// -------------------------------------------------
import WithThemeComponent from '../utils/withThemeComponent';

const SidebarWithTheme = WithThemeComponent(Sidebar);

describe('Sidebar Testing', () => {
  it('Should render the Sidebar component', () => {
    render(<SidebarWithTheme />);
    screen.getByRole('heading', {
      name: /estados/i,
    });
  });
  it('Should render the Sidebar component with max', () => {
    render(<SidebarWithTheme />);
    screen.getByText(/max/i);
  });
  it('Should render the Sidebar component with min', () => {
    render(<SidebarWithTheme />);
    screen.getByText(/min/i);
  });
  it('should match snapshot', () => {
    const component = render(<SidebarWithTheme />);
    screen.logTestingPlaygroundURL();
    expect(component).toMatchSnapshot();
  });
});
