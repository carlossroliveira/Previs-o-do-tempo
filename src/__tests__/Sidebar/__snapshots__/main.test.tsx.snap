// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Testing should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  grid-area: sidebar;
  color: #EEEEEE;
  font-family: Roboto,sans-serif;
  -webkit-transform: translateX(-50px);
  -ms-transform: translateX(-50px);
  transform: translateX(-50px);
  -webkit-animation: eQudGy 0.5s linear forwards;
  animation: eQudGy 0.5s linear forwards;
}

.c1 {
  text-align: center;
  font-size: 2rem;
  margin: 1rem 0 3rem 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 44px;
}

.c3 p + p {
  margin-left: 1rem;
}

.c3 p {
  margin: 0;
}

.c8 {
  width: 130px;
  margin-left: 2rem;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c4 {
  margin: 0;
  padding: 0;
  list-style: none;
  height: calc(100vh - 495px);
  overflow-y: scroll;
}

.c4::-webkit-scrollbar {
  width: 10px;
}

.c4::-webkit-scrollbar-thumb {
  background-color: #EEEEEE;
}

.c4::-webkit-scrollbar-track {
  border: 1px solid #EEEEEE;
  border-radius: 3px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  position: relative;
  border: 1px solid #EEEEEE;
  border-radius: 5px;
  margin: 10px 1rem;
  padding: 5px;
  cursor: pointer;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.c5 .c2 {
  margin-left: 0;
}

.c5:hover {
  opacity: 0.7;
  font-weight: 600;
  -webkit-transform: translateX(10px);
  -ms-transform: translateX(10px);
  transform: translateX(10px);
}

.c5:hover .c7 {
  overflow: visible;
  white-space: normal;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c6 {
  position: absolute;
  left: 0;
  width: 10px;
  height: 60%;
  border-radius: 1px;
  background-color: #EEEEEE;
}

.c9 {
  text-align: center;
  margin: 40px 0;
  font-size: 20px;
}

@media (max-width:870px) {
  .c5 {
    -webkit-box-pack: start;
    -webkit-justify-content: start;
    -ms-flex-pack: start;
    justify-content: start;
    padding-left: 30px;
  }
}

<body>
    <div>
      <section
        class="c0"
      >
        <h1
          class="c1"
        >
          Estados
        </h1>
        <div
          class="c2 c3"
        >
          <p>
            Min
          </p>
          <p>
            Max
          </p>
        </div>
        <ul
          class="c4"
        >
          <li
            class="c5"
          >
            <span
              class="c6"
            />
            <div
              class="c2 c3"
            >
              <p>
                °
              </p>
              <p>
                °
              </p>
            </div>
            <p
              class="c7 c8"
            />
          </li>
          <li
            class="c5"
          >
            <span
              class="c6"
            />
            <div
              class="c2 c3"
            >
              <p>
                °
              </p>
              <p>
                °
              </p>
            </div>
            <p
              class="c7 c8"
            />
          </li>
          <li
            class="c5"
          >
            <span
              class="c6"
            />
            <div
              class="c2 c3"
            >
              <p>
                °
              </p>
              <p>
                °
              </p>
            </div>
            <p
              class="c7 c8"
            />
          </li>
          <li
            class="c5"
          >
            <span
              class="c6"
            />
            <div
              class="c2 c3"
            >
              <p>
                °
              </p>
              <p>
                °
              </p>
            </div>
            <p
              class="c7 c8"
            />
          </li>
          <li
            class="c5"
          >
            <span
              class="c6"
            />
            <div
              class="c2 c3"
            >
              <p>
                °
              </p>
              <p>
                °
              </p>
            </div>
            <p
              class="c7 c8"
            />
          </li>
          <li
            class="c5"
          >
            <span
              class="c6"
            />
            <div
              class="c2 c3"
            >
              <p>
                °
              </p>
              <p>
                °
              </p>
            </div>
            <p
              class="c7 c8"
            />
          </li>
          <li
            class="c5"
          >
            <span
              class="c6"
            />
            <div
              class="c2 c3"
            >
              <p>
                °
              </p>
              <p>
                °
              </p>
            </div>
            <p
              class="c7 c8"
            />
          </li>
          <li
            class="c5"
          >
            <span
              class="c6"
            />
            <div
              class="c2 c3"
            >
              <p>
                °
              </p>
              <p>
                °
              </p>
            </div>
            <p
              class="c7 c8"
            />
          </li>
        </ul>
        <p
          class="c9"
        >
          © 
          2022
           | Carlos Oliveira
        </p>
      </section>
    </div>
  </body>,
  "container": .c0 {
  grid-area: sidebar;
  color: #EEEEEE;
  font-family: Roboto,sans-serif;
  -webkit-transform: translateX(-50px);
  -ms-transform: translateX(-50px);
  transform: translateX(-50px);
  -webkit-animation: eQudGy 0.5s linear forwards;
  animation: eQudGy 0.5s linear forwards;
}

.c1 {
  text-align: center;
  font-size: 2rem;
  margin: 1rem 0 3rem 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 44px;
}

.c3 p + p {
  margin-left: 1rem;
}

.c3 p {
  margin: 0;
}

.c8 {
  width: 130px;
  margin-left: 2rem;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c4 {
  margin: 0;
  padding: 0;
  list-style: none;
  height: calc(100vh - 495px);
  overflow-y: scroll;
}

.c4::-webkit-scrollbar {
  width: 10px;
}

.c4::-webkit-scrollbar-thumb {
  background-color: #EEEEEE;
}

.c4::-webkit-scrollbar-track {
  border: 1px solid #EEEEEE;
  border-radius: 3px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  position: relative;
  border: 1px solid #EEEEEE;
  border-radius: 5px;
  margin: 10px 1rem;
  padding: 5px;
  cursor: pointer;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.c5 .c2 {
  margin-left: 0;
}

.c5:hover {
  opacity: 0.7;
  font-weight: 600;
  -webkit-transform: translateX(10px);
  -ms-transform: translateX(10px);
  transform: translateX(10px);
}

.c5:hover .c7 {
  overflow: visible;
  white-space: normal;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c6 {
  position: absolute;
  left: 0;
  width: 10px;
  height: 60%;
  border-radius: 1px;
  background-color: #EEEEEE;
}

.c9 {
  text-align: center;
  margin: 40px 0;
  font-size: 20px;
}

@media (max-width:870px) {
  .c5 {
    -webkit-box-pack: start;
    -webkit-justify-content: start;
    -ms-flex-pack: start;
    justify-content: start;
    padding-left: 30px;
  }
}

<div>
    <section
      class="c0"
    >
      <h1
        class="c1"
      >
        Estados
      </h1>
      <div
        class="c2 c3"
      >
        <p>
          Min
        </p>
        <p>
          Max
        </p>
      </div>
      <ul
        class="c4"
      >
        <li
          class="c5"
        >
          <span
            class="c6"
          />
          <div
            class="c2 c3"
          >
            <p>
              °
            </p>
            <p>
              °
            </p>
          </div>
          <p
            class="c7 c8"
          />
        </li>
        <li
          class="c5"
        >
          <span
            class="c6"
          />
          <div
            class="c2 c3"
          >
            <p>
              °
            </p>
            <p>
              °
            </p>
          </div>
          <p
            class="c7 c8"
          />
        </li>
        <li
          class="c5"
        >
          <span
            class="c6"
          />
          <div
            class="c2 c3"
          >
            <p>
              °
            </p>
            <p>
              °
            </p>
          </div>
          <p
            class="c7 c8"
          />
        </li>
        <li
          class="c5"
        >
          <span
            class="c6"
          />
          <div
            class="c2 c3"
          >
            <p>
              °
            </p>
            <p>
              °
            </p>
          </div>
          <p
            class="c7 c8"
          />
        </li>
        <li
          class="c5"
        >
          <span
            class="c6"
          />
          <div
            class="c2 c3"
          >
            <p>
              °
            </p>
            <p>
              °
            </p>
          </div>
          <p
            class="c7 c8"
          />
        </li>
        <li
          class="c5"
        >
          <span
            class="c6"
          />
          <div
            class="c2 c3"
          >
            <p>
              °
            </p>
            <p>
              °
            </p>
          </div>
          <p
            class="c7 c8"
          />
        </li>
        <li
          class="c5"
        >
          <span
            class="c6"
          />
          <div
            class="c2 c3"
          >
            <p>
              °
            </p>
            <p>
              °
            </p>
          </div>
          <p
            class="c7 c8"
          />
        </li>
        <li
          class="c5"
        >
          <span
            class="c6"
          />
          <div
            class="c2 c3"
          >
            <p>
              °
            </p>
            <p>
              °
            </p>
          </div>
          <p
            class="c7 c8"
          />
        </li>
      </ul>
      <p
        class="c9"
      >
        © 
        2022
         | Carlos Oliveira
      </p>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
